src/python/pants/base/specs.py

```diff
@@ -104,87 +94,82 @@ def __post_init__(self) -> None:
         if self.name is None:
             raise ValueError(f"A SingleAddress must have a name. Got: {self}")
 
-    def to_spec_string(self) -> str:
-        return "{}:{}".format(self.directory, self.name)
-
-    def matching_address_families(self, address_families_dict: Dict[str, Any]) -> List[Any]:
-        return self.address_families_for_dir(address_families_dict, self.directory)
+    def __str__(self) -> str:
+        return f"{self.directory}:{self.name}"
 
-    class _SingleAddressResolutionError(Exception):
-        def __init__(self, single_address_family: Any, name: str) -> None:
-            super().__init__()
-            self.single_address_family = single_address_family
```

Comment:
We no longer raise this exception in `specs.py`, as we ended up just catching the error in `build_files.py`. Instead, the `matching_addresses()` method returns 0 or 1 address. `build_files.py` will error if it was 0.